src/main/resources/META-INF/plugin.xml

```diff
@@ -5,12 +5,42 @@
     <vendor>mitchellolsthoorn</vendor>
 
     <depends>com.intellij.modules.platform</depends>
+    <!-- add java functionality. See https://blog.jetbrains.com/platform/2019/06/java-functionality-extracted-as-a-plugin/ for more information -->
+    <depends>com.intellij.modules.java</depends>
+
+    <description>
+        <![CDATA[
+            <p>A plugin that uses EvoSuite to generate tests. Developed by TU Delft students.
+        ]]>
+    </description>
 
     <extensions defaultExtensionNs="com.intellij">
         <applicationService serviceImplementation="com.github.mitchellolsthoorn.testgenie.services.MyApplicationService"/>
         <projectService serviceImplementation="com.github.mitchellolsthoorn.testgenie.services.MyProjectService"/>
+        <toolWindow id="TestGenie" secondary="true" anchor="right"
+                    factoryClass="com.github.mitchellolsthoorn.testgenie.toolwindow.TestGenieToolWindowFactory"/>
+        <applicationService
+                serviceImplementation="com.github.mitchellolsthoorn.testgenie.settings.TestGenieSettingsService"/>
+        <applicationConfigurable
+                parentId="tools"
+                instance="com.github.mitchellolsthoorn.testgenie.settings.TestGenieSettingsConfigurable"
+                id="com.github.mitchellolsthoorn.testgenie.settings.TestGenieSettingsConfigurable"
+                displayName="TestGenie"/>
     </extensions>
 
+    <actions>
+        <!-- TestGenie action for generating tests for a class -->
```

Comment:
Should we include those types of comments in extensions part also?

Reply:
I have left it here because earlier there was another action in another group that we added into `Code` section, and it was difficult to see which action was responsible for what. However, after speaking with the client and making a proper design, we have decided that it is unnecessary and i have removed it.

It won't hurt if you add a comment. I, to be honest, just have forgotten. But as long as it is readable, this is not a problem